### PR TITLE
Support for DotCover configuration file added

### LINKED ISSUE #1401: Support for dotCover configuration file
As described in the [help](https://www.jetbrains.com/help/dotcover/2016.2/dotCover__Console_Runner_Commands.html#cover), dotCover can also be called with a `configuration file`:

    dotCover cover <configuration file name> [parameters]

I guess this is especially interesting for specifying filters. Since they might be reused from within the IDE at some point.

### PR DESCRIPTION
**Pull request by AdaskoTheBeAsT**
Support for DotCover configuration file added - fixes #1401

### CODE REVIEW

File: src/Cake.Common/Tools/DotCover/DotCoverTool.cs [hunk 1]
```diff
@@ -70,5 +70,22 @@ protected ProcessArgumentBuilder GetArguments(DotCoverSettings settings)
...
         }
+
+        /// <summary>
+        /// Get configuration full path from coverage settings
+        /// </summary>
+        /// <param name="settings">The settings</param>
+        /// <returns>The process arguments</returns>
+        protected ProcessArgumentBuilder GetConfigurationFileArgument(DotCoverSettings settings)
+        {
+            var builder = new ProcessArgumentBuilder();
```
Comment (mholo65) [inline src/Cake.Common/Tools/DotCover/DotCoverTool.cs]: Any reason why this is added to a separate method and nit part of `GetArguments()`?

Comment (AdaskoTheBeAsT) [reply src/Cake.Common/Tools/DotCover/DotCoverTool.cs]: Based on documentation for "cover" command
https://www.jetbrains.com/help/dotcover/dotCover__Console_Runner_Commands.html
"
It is possible to specify part of the parameters or all of them in the configuration file.
usage: dotCover cover <configuration file name> [parameters]
"
configuration file name should be right after command

Comment (AdaskoTheBeAsT) [reply src/Cake.Common/Tools/DotCover/DotCoverTool.cs]: @mholo65 do you have some tips how can it be done in some other way which will better fit Cake in general?

**Comment by AdaskoTheBeAsT** [comment]
Branch stalled - closing.

**Comment by patriksvensson** [comment]
@AdaskoTheBeAsT What do you mean with "Branch stalled" here?

**Comment by AdaskoTheBeAsT** [comment]
I messed up with rebasing - in pull should be only my changes. Closed for safeness.

**Pull request closed without merging**